blackjack.js

```diff
@@ -1,15 +1,39 @@
-const blackjackDeck = getDeck();
+
+
+const blackjackDeck = () =>{
+
+const deck = getDeck();
+
+let val = deck.val;
+let displayVal = getDeck().displayVal;
+let suit = getDeck().suit;
+
+allCards = ({val}, {display},{suit});
+return allCards;
+}
 
 /**
  * Represents a card player (including dealer).
  * @constructor
  * @param {string} name - The name of the player
  */
-class CardPlayer {}; //TODO
+
+class CardPlayer {
+  constructor(name){
+    this.name = name;
+    this.hand = [];
+  }
+  drawCard(){
+    //selects a card at random from the deck
+    //adds card to hand array.
+    const rdmCard = Math.floor(Math.random() * (blackjackDeck.length));
+    let playerHand = this.hand.push[rdmCard];
```

Comment:
There's a similar deal here, but results in a more usable end result. `playerHand` as a declared variable  here is not being used and disappears as soon as the method call is complete (due to lexical scoping). `this.hand.push(rdmCard)` is good though, that will write the change to `this.hand`. So you do have a syntax error there, since you are calling the `push()` method you need the `()` instead of the `[]`. Then, just delete everything else, the `=` and everything left of it on this line.